geo/services/Geo_LocationService.php

```diff
@@ -3,163 +3,63 @@
-        }
-
-        $data = array(
-            "ip"=>$data->traits->ip_address,
-            "country_code"=>$data->country->iso_code,
-            "country_name"=>$data->country->names->en,
-            "region_name"=>$regionName,
-            // Yes i know, i am not getting postcode etc yet.
-            "city"=>$city,
-            "latitude"=>$data->location->latitude,
-            "longitude"=>$data->location->longitude,
-            "cached"=>false
-        );
-
-        return $data;
-    }
-
-    // private function getTelizeData($ip){
-
-    //     $url = "/geip/".$ip;
-    //     $telizeClient = new \Guzzle\Http\Client("http://www.telize.com");
-    //     $response = $telizeClient->get($url, array(), array("exceptions" => false))->send();
-
-    //     if (!$response->isSuccessful()) {
-    //         return array();
-    //     }
-
-    //     $data = json_decode($response->getBody(),true);
-
-    //     $data = array(
-    //         "ip"=>$data['ip'],
-    //         "country_code"=>$data['country_code'],
-    //         "country_name"=>$data['country'],
-    //         "latitude"=>$data['latitude'],
-    //         "longitude"=>$data['longitude'],
-    //         "cached"=>false
-    //     );
-
-    //     return $data;
-    // }
-
-    private function anonymizeIp($ip) {
-        if (filter_var($ip, FILTER_VALIDATE_IP) !== false) {
-            $segments = explode('.', $ip);
-
-            switch (strlen($segments[3])) {
-                case 1:
-                case 2:
-                    // Set last segment to zero
-                    $segments[3] = 0;
-                    break;
-
-                case 3:
-                    // Keep the first digit and set the last two to zero
-                    $ending = substr($segments[3], 0, 1);
-                    if (substr($segments[3], 1) == '00') {
-                        // Set two random digits if IP already ended with two zeros
-                        $ending .= rand(0, 9);
-                        $ending .= rand(0, 9);
-                    } else {
-                        $ending .= '00';
-                    }
-                    $segments[3] = $ending;
-                    break;
-            }
-
-            $anonymizedIp = implode('.', $segments);
-            return $anonymizedIp;
-        } else {
-            // Not a valid IP
-            return $ip;
-        }
-    }
-
-}
+	public function getIpData()
+	{
+		$response = $this->fetchIpData();
+
+		if(!empty($response)){
+			return $response;
+		}
+	}
+
+	public function getIsEu()
+	{
+		$isEu = false;
+		$response = $this->fetchIpData();
+
+		if(!empty($response)){
+			$isEu = $response['is_eu'];
+		}
+
+		return $isEu;
+	}
+	
+	private function fetchIpData()
+	{
+		$devMode = craft()->config->get('devMode');
+		$ipApiKey = craft()->config->get('ipApiKey', 'geo');
+		$ip = craft()->request->getIpAddress();
+		$isEuUserCookie = 'craft_geo_eu_user';
+		$data = array(
+			'country_code'=>'',
+			'is_eu'=>''
+		);
```

Comment:
also this, adding a method that return all avail data or having this be configurable in the config file

Reply:
or accepting fields as a param